### DISCUSSION TITLE
Adding a comment for explaining the last statement in the `fls` function

### DISCUSSION THREAD

**Discussion by PaulRBerg** [Vectorized/solady#327] [General]
I was looking at the `fls` function in `LibBit`, and the last statement caught my attention:

https://github.com/Vectorized/solady/blob/5d0198b498601ac7ca600afb351103bac89ac0e7/src/utils/LibBit.sol#L35-L37

It would be helpful to provide an explainer comment above this line, so that readers of the contract know why you use 251, 224, and the other hardcoded values in this function.

**Comment by Vectorized** [comment]
Commenting for posterity.

`fls` uses magic numbers found via z3.

```python
from z3 import *
import math

lookup = lambda x, l: (l >> ((31 - (x & 31)) << 3)) & 0xff

l = BitVec('l', 256)
q = BitVec('q', 256)

solve(*[lookup(q >> i, l) == int(math.log2(i)) if i != 0 else lookup(q >> i, l) == 0 for i in range(256)], q >> 128 == 0, l & 0xffffffff == 0)
```

`ffs` uses magic numbers found via z3.

```python
from z3 import *

lookup = lambda x, q, l: (l >> ((31 - ((q / x if x > 0 else 0) & 31)) << 3)) & 0xff

l = BitVec('l', 256)
q = BitVec('q', 256)

solve(*[lookup(1 << i, q, l) == i for i in range(32)], lookup(0, q, l) == 0)
```
